pysb/logging.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import absolute_import
+import logging
+import platform
+import socket
+import time
+import os
+import warnings
+import pysb
+
+SECONDS_IN_HOUR = 3600
+DEBUG_ENV_VAR = 'PYSB_DEBUG'
+BASE_LOGGER_NAME = 'pysb'
+
+
+def formatter(time_utc=False):
+    """
+    Build a logging formatter using local or UTC time
+
+    Parameters
+    ----------
+    time_utc : bool, optional (default: False)
+        Use local time stamps in log messages if False, or Universal
+        Coordinated Time (UTC), also known as Greenwich Mean Time (GMT) if True
+
+    Returns
+    -------
+    A logging.Formatter object for PySB logging
+    """
+    log_fmt = logging.Formatter('%(asctime)s.%(msecs).3d - %(name)s - '
+                                '%(levelname)s - %(message)s',
+                                datefmt='%Y-%m-%d %H:%M:%S')
+    if time_utc:
+        log_fmt.converter = time.gmtime
+
+    return log_fmt
+
+
+def setup_logger(level=logging.INFO, console_output=True, file_output=False,
+                 time_utc=False, capture_warnings=True):
+    """
+    Set up a new logging.Logger for PySB logging
+
+    Calling this method will override any existing handlers, formatters etc.
+    attached to the PySB logger. Typically, :func:`get_logger` should be
+    used instead, which returns the existing PySB logger if it has already
+    been set up, and can handle PySB submodule namespaces.
+
+    Parameters
+    ----------
+    level : int
+        Logging level, typically using a constant like logging.INFO or
+        logging.DEBUG
+    console_output : bool
+        Set up a default console log handler if True (default)
+    file_output : string
+        Supply a filename to copy all log output to that file, or set to
+        False to disable (default)
+    time_utc : bool
+        Specifies whether log entry time stamps should be in local time
+        (when False) or UTC (True). See :func:`formatter` for more
+        formatting options.
+    capture_warnings : bool
+        Capture warnings from Python's warnings module if True (default)
+
+    Returns
+    -------
+    A logging.Logger object for PySB logging. Note that other PySB modules
+    should use a logger specific to their namespace instead by calling
+    :func:`get_logger`.
+    """
+    log = logging.getLogger(BASE_LOGGER_NAME)
+
+    # Logging level can be overridden with environment variable
+    if DEBUG_ENV_VAR in os.environ:
```

Comment:
The env var should allow specific levels, not just DEBUG, especially if the default is changed to WARNING (might be nice to get INFO level sometimes but not full DEBUG). Possibly the var name could change to PYSB_VERBOSE and it could take integer levels or string names (same as the level names from `logging`).